eng/native/configurecompiler.cmake

```
@@ -787,6 +787,13 @@ if (CLR_CMAKE_HOST_WIN32)
     endif()
 
 elseif (NOT CLR_CMAKE_HOST_BROWSER)
+    # This is a workaround for upstream issue: https://gitlab.kitware.com/cmake/cmake/-/issues/22995.
+    #
+    # In Clang.cmake, the decision to use single or double hyphen for target and gcc-toolchain is made
+    # based on CMAKE_${LANG}_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN, but CMAKE_ASM_COMPILER_VERSION is empty
+    # so it picks up single hyphen options, which new clang versions don't recognize.
```

```suggestion
    # In Clang.cmake, the decision to use single or double hyphen for target and gcc-toolchain
    # is made based on CMAKE_${LANG}_COMPILER_VERSION, but CMAKE_ASM_COMPILER_VERSION is empty
    # so it picks up single hyphen options, which new clang versions don't recognize.
```
https://github.com/Kitware/CMake/blob/d6fbfc94a1923c3a92b97ad7028a77d4f0f850d2/Modules/Compiler/Clang.cmake#L37 (the cross compilation was failing with `clang: error: unknown argument: '-gcc-toolchain'` when host has latest llvm-toolchain v15 installed; found with `ubuntu-22.04-cross-riscv64-20220818214620-4d4c24f` image).